spec/helpers/i18n_helper_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+RSpec.describe I18nHelper do
+  describe '#date_eras_gregorian' do
+    context 'before christ' do
+      it 'converts gregorian before christ to correct format' do
+        ['-1234', ' -1234', '-1234 ', ' -1234 ', '- 1234', ' - 1234', '- 1234 ', ' - 1234 '].each do |date|
+          formatted_date = helper.date_eras_gregorian(date)
+          expect(formatted_date).to eq('1234 BCE'), "date = '#{date}' has failed"
```

Comment:
I see you added '12' to the list, but according to the specs it think that it should not be converted since it doesn't have a '+' or '-', we can discuss later.